### PR TITLE
PSI4 version with maxam=7

### PR DESCRIPTION
**Pull request by wpoely86**
requires #1854

### CODE REVIEW

File: easybuild/easyconfigs/p/PSI/PSI-4.0b6-20150814-intel-2015a-mt-maxam7.eb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+python = 'Python'
+pyver = '2.7.10'
+
+dependencies = [
+    (python, pyver),
+    ('Boost', '1.59.0', '-%s-%s' % (python, pyver)),
+]
+
+builddependencies = [
+    ('CMake', '3.2.3'),
```
Comment (boegel) [inline easybuild/easyconfigs/p/PSI/PSI-4.0b6-20150814-intel-2015a-mt-maxam7.eb]: bump version?

easyconfig for 3.2.3 is there (3.3.1 is latest)

Comment (boegel) [inline easybuild/easyconfigs/p/PSI/PSI-4.0b6-20150814-intel-2015a-mt-maxam7.eb]: see #1873

**Comment by boegel** [comment]
Jenkins: test this please

File: easybuild/easyconfigs/p/PSI/PSI-4.0b6-20150814-intel-2015a-mt-maxam7.eb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    'PSI-%(version)s-fix-gitversion.patch',
+]
+
+configopts = '-DLIBINT_OPT_AM=7 -DENABLE_PLUGINS=ON -DENABLE_DUMMY_PLUGIN=ON '
+
+python = 'Python'
+pyver = '2.7.10'
+
+dependencies = [
+    (python, pyver),
```
Comment (boegel) [inline easybuild/easyconfigs/p/PSI/PSI-4.0b6-20150814-intel-2015a-mt-maxam7.eb]: `-Python-2.7.10` should be in the versionsuffix too?

cfr. #1855

Comment (wpoely86) [inline easybuild/easyconfigs/p/PSI/PSI-4.0b6-20150814-intel-2015a-mt-maxam7.eb]: mmm, I really don't like this very long versionsuffixes. It hard to read and makes them pointless.

Comment (boegel) [inline easybuild/easyconfigs/p/PSI/PSI-4.0b6-20150814-intel-2015a-mt-maxam7.eb]: see https://github.com/wpoely86/easybuild-easyconfigs/pull/27

Comment (boegel) [inline easybuild/easyconfigs/p/PSI/PSI-4.0b6-20150814-intel-2015a-mt-maxam7.eb]: For commonly used things like Python, it really helps in trying to get users to load modules that are compatible with each other.

But, I agree this is not an ideal solution.

A cleaner way would be to use a module hierarchy and have Python as one of the hierarchy layers. But then you run into the 'matrix' problem, which is yet to be solved...

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/50d5152b4e0afdadab15 for a full test report.

**Comment by boegel** [comment]
@wpoely86: hmm, maybe Boost 1.59.0 is not such a good idea after all...

```
/gpfs/scratch/projects/project_gpilot/vsc40023/software/Boost/1.59.0-intel-2015a-Python-2.7.10/include/boost/python/converter/builtin_converters.hpp(172): error: namespace "std" has no member "complex"
  BOOST_PYTHON_TO_PYTHON_BY_VALUE(std::complex<float>, ::PyComplex_FromDoubles(x.real(), x.imag()), &PyComplex_Type)
```

**Comment by wpoely86** [comment]
Again complex? It worked with 1.58 for me.

**Comment by boegel** [comment]
Testing with 1.58.0 on my end, I'll keep you posted (but feel free to downgrade the Boost dep back to 1.58.0 already in here).

**Comment by boegel** [comment]
@wpoely86: I'm running into exactly the same issue with Boost 1.58.0...

**Comment by wpoely86** [comment]
https://gist.github.com/e327cb33c52d5d82e227 works for me. Odd.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/074a4e65cfe954b6040a for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2630L v3 @ 1.80GHz, Python 2.7.5
See https://gist.github.com/a892ed6970ca11fdb610 for a full test report.

**Comment by boegel** [comment]
Works with CMake v3.2.3 just fine... Good to know that CMake 3.3.x may cause problems like this.

**Comment by boegel** [comment]
Thanks @wpoely86!

**Pull request merged**